File: example/.vscode/l.js
const fs = require('fs');
const readline = require('readline');
const filePath = process.argv[2];
const scheme = process.argv[3];

console.log(`PROJECT_SCEME: ${scheme}`)

let lastKnownPosition = 0;

function filterLine(line) {
    if (line.includes("com.apple."))
        return null;
    if (line.includes("   Activity    "))
        return null;

    const parts = line.split(scheme);

    // Extract the portion after "TestVSCode:"
    const result = parts.length > 1 ? parts[1] : line;

    return result;
}

function printNewLines() {
    const fileStream = fs.createReadStream(filePath, { start: lastKnownPosition });
    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity,
    });

    rl.on('line', (line) => {
        toTrack = filterLine(line);
        if (toTrack !== null) {
            console.log(`${toTrack}\n`);
        }

        lastKnownPosition += line.length + 1; // Add 1 for the newline character
    });

    rl.on('close', () => {
        //console.clear(); // Optional: Clear the console before printing to simulate an update
    });
}

// Watch for changes in the file
fs.watchFile(filePath, (curr, prev) => {
    if (curr.mtime > prev.mtime) {
        printNewLines();
    }
});

function watchFile(filepath, oncreate, ondelete, onchange) {
    var
        fs = require('fs'),
        path = require('path'),
        filedir = path.dirname(filepath),
        filename = path.basename(filepath);

    fs.watch(filedir, function (event, who) {
        if (event === 'rename' && who === filename) {
            if (fs.existsSync(filepath)) {
                oncreate();
            } else {
                ondelete();
            }
        } else if (event == 'change' && who == filename) {
            onchange();
        }
    })
}

watchFile('.vscode/log.changed', function () { }, function () { }, function () {
    console.clear();
    lastKnownPosition = 0;
    console.log('RELAUNCHING APPLICATION...');
})

// Initial print
printNewLines();

File: .vscode/l.py
import os
import time
import sys

file_path = sys.argv[1]
project_scheme = os.environ.get('PROJECT_SCHEME')

last_known_position = 0

def filter_line(line):
    return line


def print_new_lines():
    global last_known_position
    with open(file_path, 'r') as file:
        file.seek(last_known_position)
        for line in file:
            to_track = filter_line(line.strip())
            if to_track:
                print(f"{to_track}")

            last_known_position += len(line) + 1  # Add 1 for the newline character

        sys.stdout.flush()


# Watch for changes in the file
def watch_file(filepath, on_delete, on_change):
    filedir, filename = os.path.split(filepath)

    stat = os.path.getmtime(filepath)
    while True:
        print_new_lines()
        time.sleep(1)
        try:
            if stat < os.path.getmtime(filepath):
                stat = os.path.getmtime(filepath)
                on_change()
        except FileNotFoundError:
            on_delete()
            continue


def on_delete():
    print(f'Log is deleted for Application {project_scheme}')


def on_change():
    global last_known_position
    os.environ['TERM'] = 'xterm'  # Set the TERM variable to a reasonable default
    os.system('clear')  # Clear the console before printing to simulate an update
    last_known_position = 0
    print(f'RELAUNCHING {project_scheme} APPLICATION...')


# Watch for changes in the file
watch_file('.vscode/log.changed', on_delete, on_change)